consul/query_endpoint.go

```diff
@@ -0,0 +1,454 @@
+package consul
+
+import (
+	"errors"
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/armon/go-metrics"
+	"github.com/hashicorp/consul/consul/structs"
+)
+
+var (
+	// ErrQueryNotFound is returned if the query lookup failed.
+	ErrQueryNotFound = errors.New("Query not found")
+)
+
+// Query manages the prepared query endpoint.
+type Query struct {
```

Comment:
I was wondering what you would think about this :-) It felt ok but I agree it's pretty overloaded. I'll change it to `PreparedQuery` everywhere which I pretty clear.